### PR TITLE
Done all the Requirements

### PR DESCRIPTION
**Pull request by shane777**
Project can download and run in local, all the documents write in "Readme.md". Thanks.

### CODE REVIEW

File: src/tableRow.js [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+                      return b[type] - a[type];
+                  case 'name' :
+                      return a[type].toString().localeCompare(b[type].toString());
+                  default :
+                      return 0;
+              }
+          }
+      )
+  }
+  containSearch (searchValue){
```
Comment (flyingant) [inline src/tableRow.js]: hhhhh, do you have any better idea for this function?
e.g. combine all the attributes to a string and then search the string.(a+b+c+d).indexOf(searchValue)

**Comment by flyingant** [comment]
Hey zhang，

I‘m MaYi and I'm the guy who reviewed your code.
You are doing well and I would like to challenge you in order to know more about you ability.

1. I noticed that you build your project by `react-script` and your CV mentioned you had some experiences on `webpack`. So could you try to rebuild your project by `webpack`?
2. Could you refactor your CSS with `LESS` or `SCSS`?
3. Your react skill seems solid and maybe you could try to add more cool things to your project. Could you try to add `redux` help to manage the data? (Notes: even though this proejct is too tiny and no need to use `redux` but I hope to know more about you.)
4. Trying to add something you think is cool....

Leave your message here if you got any questions.

MaYi

**Comment by shane777** [comment]
I will try it and commit it in today. Thanks.

**Comment by flyingant** [comment]
Cool. 
No hurries but make it better.

**Comment by shane777** [comment]
Hi, I use webpack to build it. and the output file is in public/build.js. You can run the project by click index.html.
I am really sorry that maybe I have no enough time to do all the task because it is my first time in react project and have no redux experience before. I saw the document about Redux, it not difficult, may takes me 1-2 days to restructure the data of this project but I have another offer deadline at tomorrow, I want to know will I have opportunity to join you if I done all the task? So I can decide reject that offer or not.
Really sorry for my hurry and hope can work with you.
Thanks.

**Comment by makara** [comment]
Hey @shane777. Short answer, there's always the risk and you'll need to be confident enough to take it. Longer answer:

- We are not challenging people for no reason. If you can handle it well we'll be happy to have you on board, but that depends on the result and we cannot guarantee it.
- A challenge is a plus. You are free to refuse further challenges and we could already take a decision.

**Comment by shane777** [comment]
Got it. I will do the task good and ASAP. It looks not difficult for me to do this decision because I want to join you for a long time. Wait for my pr and thanks.

**Comment by flyingant** [comment]
Fantastic. @shane777 Feel free to ask any question if you got any problem.

**Comment by shane777** [comment]
Hi , I have done all the task that rebuild this project with scss and redux. And deploy it on git pages, path is https://shane777.github.io/test-frontend/ . Thanks.

**Comment by makara** [comment]
Thanks @shane777. We'll get back to you soon. Closing.

**Pull request closed without merging**